src/sonic_ax_impl/mibs/vendor/cisco/ciscoSwitchQosMIB.py

```diff
@@ -130,6 +135,46 @@ def update_stats(self):
                         self.mib_oid_list.append(mib_oid)
                         self.mib_oid_to_queue_map[mib_oid] = counter_value
 
+        # Update queue counters for LAG
+        lag_range = sorted(list(self.oid_lag_name_map.keys()))
+        for lag_index in lag_range:
+            lag_oid_list = []
+            lag_oid_to_queue_map = {}
+            # Get counters for each LAG member
+            for lag_member in self.lag_name_if_name_map[self.oid_lag_name_map[lag_index]]:            
+                lag_member_queues = []
+                try:
+                    lag_member_queues = self.port_queue_list_map[mibs.get_index(lag_member)]
+                except KeyError:
+                    # LAG member does not has a queues, continue..
+                    continue
+
+                for queue in lag_member_queues:
+                    # Get queue type and statistics
+                    queue_sai_oid = self.port_queues_map[mibs.queue_key(mibs.get_index(lag_member), queue)]
+                    queue_stat_table_name = mibs.queue_table(queue_sai_oid)
+                    queue_type = self.queue_type_map.get(queue_sai_oid)
+                    queue_stat = self.queue_stat_map.get(queue_stat_table_name, {})
+
+                    # Add supported counters to MIBs list and store counters values
+                    for (counter, counter_type), counter_mib_id in CounterMap.items():
+                        # Only egress queues are supported
+                        mib_oid = (lag_index, int(DirectionTypes.EGRESS), queue + 1, counter_mib_id)
+
+                        counter_value = 0
+                        if queue_type == counter_type:
+                            counter_value = int(queue_stat.get(counter, 0))
+
+                            if mib_oid not in lag_oid_list:
+                                lag_oid_list.append(mib_oid)
+                                lag_oid_to_queue_map[mib_oid] = counter_value
+                            else:
+                                lag_oid_to_queue_map[mib_oid] += counter_value
+
+            # Add LAG port counters to MIB
+            self.mib_oid_list += lag_oid_list
```

Comment:
>+= [](start = 30, length = 2)

Are you assuming lag mib_oid is greater than port mib_oid? Could you remove this assumption?

Reply:
Sorry, I don't get your comment.
I have 2 different objects: one for queue counters of each port, second for lag member of each LAG port. I read lag members and calculate the queue counters for each of them. 
There is similar implementation for LAG in the "InterfacesUpdater" of rfc1213.py.

Reply:
self.mib_oid_list is a sorted list. So you are assuming element in lag_oid_list is larger than element in self.mib_oid_list, right?

---
In reply to: [158643144](https://github.com/andrii-savka/sonic-snmpagent/pull/5#discussion_r158643144) [](ancestors = 158643144)

Reply:
Yes, elements in lag_oid_list are larger than element in self.mib_oid_list. 
Do you mean I need to re-sort the elements in the list ?

Reply:
I am just saying 'hidden assumption' is not good, considering readability and easy extension. One quick solution is to remove previous sorting completely, and do one sort at the end.

---
In reply to: [158652615](https://github.com/andrii-savka/sonic-snmpagent/pull/5#discussion_r158652615) [](ancestors = 158652615)

Reply:
@qiluo-msft 
Added more comments because it's actually the best solution from my point of view. 
self.mib_oid_list contains a OID in a specific order and can't be sorted (or it will be heavy from performance point of view). 
we need to sort port and lag lists and then make a OID list in specific order based on sorted ports range. 
Also we need to calculate the counters for LAG ports. 
It would very thankful if you suggest better solution.

Reply:
for if_index in self.oid_sai_map:
    ...
    self.mib_oid_list.append(mib_oid)

for lag_index in self.oid_lag_name_map:
    ...
    self.mib_oid_list.append(mib_oid)

sort(self.mib_oid_list)


---
In reply to: [158733677](https://github.com/andrii-savka/sonic-snmpagent/pull/5#discussion_r158733677) [](ancestors = 158733677)